lua/telescope/_extensions/zk.lua

```diff
@@ -77,8 +83,45 @@ local telescope_zk_notes = function(opts)
     }):find()
 end
 
+local telescope_zk_grep = function(opts)
+    opts = opts or {}
+
+    local cwd = _G.zk_config.default_notebook_path
+    opts.cwd = opts.cwd or cwd
+    opts.cwd = opts.cwd and vim.fn.expand(opts.cwd) or vim.loop.cwd()
+    opts.entry_maker = create_entry_maker()
+
+    local zk_notes_grep = finders.new_job(function(prompt)
+        if not prompt or prompt == "" then
+            return { "zk", "list", "--footer", "\n", "-q", "-P", "--format", "{{ path }}\t{{ title }}" }
```

Comment:
@pstuifzand You're using some new flags that are not yet released in `zk`, so I'll wait for the next release before merging your PR. I'll probably do a release this week-end.

Reply:
Alright. The footer flag is needed to get the last result. Telescope seems to have an issue where only lines ending in newline are added. The footer flag adds the last newline to make this work.

I guess the last line doesn't have a newline by default, because it makes formatting easier?

Reply:
> I guess the last line doesn't have a newline by default, because it makes formatting easier?

It makes it more flexible, the newline is added **between** items with `--delimiter \n` by default, not after each item. Combined with `--header` and `--footer` you can create valid JSON objects:

```
--delimiter , --header { --footer }
```

Reply:
I have written a test case for telescope that shows the issue there.

The --footer option solves the issue here. Perhaps it makes sense to have newline be the default value for footer. Most command line tools output a new line after every line. Even with the change in default your example should work the same.

Reply:
Yeah I think that would be a good idea, I'll make `\n` the default value.

Reply:
`--footer` and `--header` are released in https://github.com/mickael-menu/zk/releases/tag/v0.6.0

I forgot to make `\n` the default value though, so it will be for the next release.